src/platform/linux/mod.rs

```diff
@@ -17,27 +17,53 @@
 pub mod sys;
 
 mod device;
-pub use self::device::{Device, Queue};
+pub use self::device::Device;
 
-use crate::configuration::Configuration as C;
-use crate::error::*;
+use crate::configuration::Configuration;
+use crate::error::Result;
 
 /// Linux-only interface configuration.
-#[derive(Copy, Clone, Default, Debug)]
-pub struct Configuration {
+#[derive(Copy, Clone, Debug)]
+pub struct PlatformConfig {
+    /// switch of Enable/Disable packet information for network driver
     pub(crate) packet_information: bool,
+    /// root privileges required or not
+    pub(crate) ensure_root_privileges: bool,
 }
 
-impl Configuration {
-    /// Enable or disable packet information, when enabled the first 4 bytes of
-    /// each packet is a header with flags and protocol type.
+/// `packet_information` is default to be `false` and `ensure_root_privileges` is default to be `true`.
+impl Default for PlatformConfig {
+    fn default() -> Self {
+        PlatformConfig {
+            packet_information: false,
+            ensure_root_privileges: true,
+        }
+    }
+}
+
+impl PlatformConfig {
+    /// Enable or disable packet information, the first 4 bytes of
+    /// each packet delivered from/to Linux underlying API is a header with flags and protocol type when enabled.
+    ///
+    /// [Note: This configuration just applies to the Linux underlying API and is a no-op on tun2(i.e. the packets delivered from/to tun2 always contain no packet information) -- end note].
+    #[deprecated(
```

Comment:
The `packet_information` switch is active and will affect the format of data packets generated by `Linux` `TUN`. 

But the `tun` crate will determine whether it contains `packet_information`. 
If so, the `tun` crate will remove or add back it; if not, it will do nothing.